velox/functions/lib/Re2Functions.cpp

```diff
@@ -294,14 +315,20 @@ class Re2SearchAndExtractConstantPattern final : public VectorFunction {
     // accommodate the largest group id referenced.
     exec::LocalDecodedVector groupIds(context, *args[2], rows);
     T maxGroupId = 0, minGroupId = 0;
-    rows.applyToSelected([&](vector_size_t i) {
+    context.applyToSelectedNoThrow(rows, [&](vector_size_t i) {
       maxGroupId = std::max(groupIds->valueAt<T>(i), maxGroupId);
       minGroupId = std::min(groupIds->valueAt<T>(i), minGroupId);
     });
-    checkForBadGroupId(maxGroupId, re_);
-    checkForBadGroupId(minGroupId, re_);
+    try {
+      checkForBadGroupId(maxGroupId, re_);
+      checkForBadGroupId(minGroupId, re_);
+    } catch (const std::exception& e) {
+      context.setErrors(rows, std::current_exception());
```

Comment:
It looks like when some rows have invalid group IDs we fail all rows. Is this intentional?

Reply:
Documentation says : 
"If the pattern is invalid or the group id is out of range, throws an
 exception. If the pattern does not match, returns null. "
 I think I will add the check / row then, this will however have some performance impact.

Reply:
This behavior would be OK without TRY, but needs to be fixed for TRY.